source/CCustomOpcodeSystem.cpp

```diff
@@ -265,6 +269,123 @@ namespace CLEO
 		return (callbackResult != OR_NONE) ? callbackResult : result;
 	}
 
+	OpcodeResult CCustomOpcodeSystem::CallFunctionGeneric(WORD opcode, CRunningScript* thread, bool thisCall, bool returnArg)
+	{
+		void* func; *thread >> func;
+		void* struc = nullptr; if(thisCall) *thread >> struc;
+		DWORD numParams; *thread >> numParams;
+		DWORD stackAlign; *thread >> stackAlign; // pop
+
+		if ((size_t)func <= CCustomOpcodeSystem::MinValidAddress)
+		{
+			SHOW_ERROR("Invalid '0x%X' function pointer param of opcode [%04X] in script %s\nScript suspended.", func, opcode, ((CCustomScript*)thread)->GetInfoStr().c_str());
+			return CCustomOpcodeSystem::ErrorSuspendScript(thread);
+		}
+
+		if (thisCall && (size_t)struc <= CCustomOpcodeSystem::MinValidAddress)
+		{
+			SHOW_ERROR("Invalid '0x%X' struct pointer param of opcode [%04X] in script %s\nScript suspended.", struc, opcode, ((CCustomScript*)thread)->GetInfoStr().c_str());
+			return CCustomOpcodeSystem::ErrorSuspendScript(thread);
+		}
+
+		int nVarArg = GetVarArgCount(thread);
+		if (numParams + returnArg != nVarArg) // and return argument
+		{
+			SHOW_ERROR("Opcode [%04X] declared %d input args, but provided %d in script %s\nScript suspended.", opcode, numParams, (int)nVarArg - returnArg, ((CCustomScript*)thread)->GetInfoStr().c_str());
+			return CCustomOpcodeSystem::ErrorSuspendScript(thread);
+		}
+
+		constexpr size_t Max_Args = 50;

```

Comment:
why 50? should be 32

Reply:
It was like that in original version.

Reply:
if it was 50, we could keep this number. To be honest CLEO Redux caps them at 16. I don't know functions with higher arity. 

as you already changed to 32 let is stay this way